File: 125-valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        String st = s.replaceAll("[^a-zA-Z0-9]","").toLowerCase();

        int i=0, j=st.length()-1;
        while(i<j){
            if(st.charAt(i)!=st.charAt(j)){
                return false;
            }
            i++;
            j--;
        }
        return true;
    }
}

File: 125-valid-palindrome/valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        # initialize two pointers
        left, right = 0, len(s) - 1

        #run the loop for pointers
        while left < right:
            #skip non-alphanumeric
            while left < right and not s[left].isalnum():
                left += 1
            while left < right and not s[right].isalnum():
                right -= 1
            
            #compare under same case
            if s[left].lower() != s[right].lower():
                return False
            
            left += 1
            right -= 1
        
        return True